controllers/remove_process_groups.go

```diff
@@ -276,21 +281,42 @@ func includeProcessGroup(ctx context.Context, r *FoundationDBClusterReconciler,
 	return nil
 }
 
-func getProcessesToInclude(cluster *fdbv1beta2.FoundationDBCluster, removedProcessGroups map[fdbv1beta2.ProcessGroupID]bool) []fdbv1beta2.ProcessAddress {
+func getProcessesToInclude(logger logr.Logger, cluster *fdbv1beta2.FoundationDBCluster, removedProcessGroups map[fdbv1beta2.ProcessGroupID]bool, adminClient fdbadminclient.AdminClient, status *fdbv1beta2.FoundationDBStatus) ([]fdbv1beta2.ProcessAddress, error) {
 	fdbProcessesToInclude := make([]fdbv1beta2.ProcessAddress, 0)
 
 	if len(removedProcessGroups) == 0 {
-		return fdbProcessesToInclude
+		return fdbProcessesToInclude, nil
+	}
+
+	excludedServers, err := adminClient.GetExclusionsFromStatus(status)
+	if err != nil {
+		return fdbProcessesToInclude, fmt.Errorf("unable to get excluded servers from status, %w", err)
+	}
+	excludedServersMap := make(map[string]fdbv1beta2.None, len(excludedServers))
+	for _, excludedServer := range excludedServers {
+		excludedServersMap[excludedServer.String()] = fdbv1beta2.None{}
 	}
 
 	idx := 0
 	for _, processGroup := range cluster.Status.ProcessGroups {
 		if processGroup.IsMarkedForRemoval() && removedProcessGroups[processGroup.ProcessGroupID] {
-			if cluster.UseLocalitiesForExclusion() {
-				fdbProcessesToInclude = append(fdbProcessesToInclude, fdbv1beta2.ProcessAddress{StringAddress: processGroup.GetExclusionString()})
+			foundInExcludedServerList := false
+			exclusionString := processGroup.GetExclusionString()
+			if _, ok := excludedServersMap[exclusionString]; ok {
+				fdbProcessesToInclude = append(fdbProcessesToInclude, fdbv1beta2.ProcessAddress{StringAddress: exclusionString})
+				foundInExcludedServerList = true
 			}
 			for _, pAddr := range processGroup.Addresses {
-				fdbProcessesToInclude = append(fdbProcessesToInclude, fdbv1beta2.ProcessAddress{IPAddress: net.ParseIP(pAddr)})
+				if _, ok := excludedServersMap[pAddr]; ok {
+					fdbProcessesToInclude = append(fdbProcessesToInclude, fdbv1beta2.ProcessAddress{IPAddress: net.ParseIP(pAddr)})
+					foundInExcludedServerList = true
+				}
+			}
+			if !foundInExcludedServerList {
+				// This means that the process is marked for exclusion and is also removed in the previous step but is missing
+				// its entry in the excluded servers in the status. This should not throw an error as this will block the
+				// inclusion for other processes, but we should have a record of this event happening in the logs.
+				logger.Error(fmt.Errorf(""), "processGroup is included but is missing from excluded server list", "processGroup", processGroup)
```

Comment:
It's probably good enough to print this out at the `info` level, as this is not an error.